api/src/routes/donate.ts

```diff
@@ -214,3 +201,164 @@ export const donateRoutes: FastifyPluginCallbackTypebox = (
 
   done();
 };
+
+/**
+ * Plugin for the donation endpoints.
+ *
+ * @param fastify The Fastify instance.
+ * @param _options Options passed to the plugin via `fastify.register(plugin, options)`.
+ * @param done The callback to signal that the plugin is ready.
+ */
+export const chargeStripeRoute: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  // Stripe plugin
+  const stripe = new Stripe(STRIPE_SECRET_KEY, {
+    apiVersion: '2024-06-20',
+    typescript: true
+  });
+
+  // @ts-expect-error - @fastify/csrf-protection needs to update their types
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  fastify.addHook('onRequest', fastify.csrfProtection);
```

Comment:
It seems very strange to have CSRF protection when the routes don't require authentication. The point being to reject requests that are authorised, but do not originate on our site. These requests don't require authorization, though.

Can we require it? If not, I don't think it makes sense to include CSRF protection.

Reply:
the wallets are currently available for unauthorized users. So we don't have to enforce authentication at this point. I can look into our data to see what percentage of the wallet users are signed in and make the call based on that.

Reply:
Yes, it should only be called from our client.

Reply:
Okay, cool. I've pushed a commit to remove this. Now `/donate/charge-stripe` and `/donate/create-stripe-payment-intent` are both accessible to non-users.